internal/provider/resource_virtual_machine_group.go

```diff
@@ -0,0 +1,131 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/krystal/go-katapult/pkg/katapult"
+)
+
+func resourceVirtualMachineGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVirtualMachineGroupCreate,
+		ReadContext:   resourceVirtualMachineGroupRead,
+		UpdateContext: resourceVirtualMachineGroupUpdate,
+		DeleteContext: resourceVirtualMachineGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"segregate": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceVirtualMachineGroupCreate(
+	ctx context.Context,
+	d *schema.ResourceData,
+	meta interface{},
+) diag.Diagnostics {
+	m := meta.(*Meta)
+	segregate := d.Get("segregate").(bool)
+	args := &katapult.VirtualMachineGroupCreateArguments{
+		Name:      d.Get("name").(string),
+		Segregate: &segregate,
+	}
+
+	vmg, _, err := m.Client.VirtualMachineGroups.Create(
+		ctx, m.OrganizationRef(), args,
+	)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_ = d.Set("name", vmg.Name)
+	_ = d.Set("segregate", vmg.Segregate)
```

Comment:
We don't actually need to set these attributes, as they'll be overwritten by the call to `resourceVirtualMachineGroupRead()` below.

```suggestion
```